clients/drcachesim/tracer/raw2trace.h

```diff
@@ -1442,6 +1455,75 @@ template <typename T> class trace_converter_t {
         return "";
     }
 
+#ifdef BUILD_PT_POST_PROCESSOR
+#    define PT_DATA_FILE_NAME_SUFFIX ".pt"
+#    define PT_METADAT_FILE_NAME_SUFFIX ".metadata"
+#    define KERNEL_IMAGE_FILE_NAME "kimage"
+#    define KERNEL_IMAGE_METADATA_FILE_NAME "kimage.metadata"
+    std::string
+    append_syscall_pt_trace(void *tls, thread_id_t thread_id, uint64_t syscall_id)
+    {
+        std::string error = "";
+        pt2ir_config_t config = {};
+        config.raw_file_path = impl()->get_syscall_pt_trace_dir(tls) + "/" +
+            std::to_string(thread_id) + "." + std::to_string(syscall_id) +
+            PT_DATA_FILE_NAME_SUFFIX;
+        config.kernel_image_path =
+            impl()->get_syscall_pt_trace_dir(tls) + "/" + KERNEL_IMAGE_FILE_NAME;
+        config.kernel_image_metadata_path =
+            impl()->get_syscall_pt_trace_dir(tls) + "/" + KERNEL_IMAGE_METADATA_FILE_NAME;
+        config.init_with_metadata(config.raw_file_path + PT_METADAT_FILE_NAME_SUFFIX);
+
+        std::unique_ptr<pt2ir_t> ptconverter(new pt2ir_t());
```

Comment:
This can just be a local var on the stack -- I don't see why this needs to be on the heap?